library/blob.c

```diff
@@ -1057,6 +1053,83 @@ static int eblob_blob_iterator(struct eblob_iterate_priv *iter_priv)
 	return err;
 }
 
+
+/**
+ * eblob_disk_control_cmp_by_position() - compares dc's by position in a blob file
+ */
+static int eblob_disk_control_cmp_by_position(const void *lhs, const void *rhs) {
+	const struct eblob_disk_control *left = (const struct eblob_disk_control *)lhs;
+	const struct eblob_disk_control *right = (const struct eblob_disk_control *)rhs;
+	if (left->position == right->position)
+		return 0;
+
+	return left->position < right->position ? -1 : 1;
+}
+
+/**
+ * eblob_blob_iterate_by_pos() - read whole index, sort it by position and run iteration over it.
+ */
+static int eblob_blob_iterate_by_pos(struct eblob_iterate_priv *iter_priv)
+{
+	struct eblob_iterate_control *ctl = iter_priv->ctl;
+	struct eblob_base_ctl *bctl = ctl->base;
+
+	struct eblob_disk_control *dc = NULL;
+	struct eblob_iterate_local loc;
+	int err = 0;
+
+	static const int hdr = sizeof(struct eblob_disk_control);
+
+	if (iter_priv == NULL) {
```

Comment:
herre you have check after usage on line 1074

Reply:
fixed